### DISCUSSION TITLE
User-assigned IDs in code snippets should be honored by the line highlighter

### DISCUSSION THREAD

**Discussion by hellt** [squidfunk/mkdocs-material#6415] [Ideas]
I am in love with the way one can link to certain line numbers from the markdown:

```
Here we [handle the `version`](#__codelineno-8-2:9) CLI flag.
```

The problem I am having is that `codelineno-8-2:9` is not a persisten id of the code block element.

If we add a code block before the one in question, its ID will auto-increment and the link won't work anymore.

Since pymarkdownx snippet extension allows to assign user-defined ID it is possible to have a static reference to the code block, that is agnostic to the document layout.

````
``` {.python #myid linenums="1"}
import bar
```
````

The problem is that `#myid` id assigned to the code block is not "respected" by the mkdocs-material, so that when one uses `#__codelineno-myid-7` the relevant line does not get highlighted, although the span elements have that ID.

It would be great to have this feature.

**Comment by facelessuser** [comment]
Then a request for material to support this would be needed. They are the ones doing the highlighting.

**Reply by facelessuser** [reply]
Sorry, I thought I was responding on my own repo 🙃.

**Reply by facelessuser** [reply]
For context, the original conversation was occurring here: https://github.com/facelessuser/pymdown-extensions/discussions/2258#discussioncomment-7740961

**Comment by squidfunk** [comment]
Might be a bug. Please create an issue with a minimal reproduction.

**Reply by hellt** [reply]
opened https://github.com/squidfunk/mkdocs-material/issues/6448
